packages/bootstrap/extra/dcos_internal_utils/exhibitor/__init__.py

```diff
@@ -44,3 +101,9 @@ def wait(master_count_filename):
     if serving != cluster_size or leaders != 1:
         msg = 'Expected {} servers and 1 leader, got {} servers and {} leaders'.format(cluster_size, serving, leaders)
         raise Exception(msg)
+
+    # Local Zookeeper is up. Config should be stable, local zookeeper happy. Stash the PID so if
+    # there is a restart we can come up quickly without requiring a new zookeeper quorum.
+    zk_pid_mtime = str(get_zk_pid_mtime())
+    log.info('Stashing zk.pid mtime %s to %s', zk_pid_mtime, stash_zk_pid_stat_mtime_path)
+    write_string(stash_zk_pid_stat_mtime_path, zk_pid_mtime)
```

Comment:
They all write out the same mtime in the absolute worse case. It should always be eventually consistent. Each one is doing a more or less atomic full file write.